libretroshare/src/retroshare/rsgxschannels.h

```diff
@@ -138,180 +199,186 @@ class RsGxsChannels: public RsGxsIfaceHelper, public RsGxsCommentService
+	virtual bool turtleChannelRequest(
+	        const RsGxsGroupId& channelId,
+	        const std::function<void (const RsGxsChannelGroup& result)>& multiCallback,
+	        rstime_t maxWait = 300 ) = 0;
+
+
+	/* Following functions are deprecated as they expose internal functioning
+	 * semantic, instead of a safe to use API */
+
+	RS_DEPRECATED_FOR(getChannelsInfo)
+	virtual bool getGroupData(const uint32_t &token, std::vector<RsGxsChannelGroup> &groups) = 0;
+
+	RS_DEPRECATED_FOR(getChannelsContent)
+	virtual bool getPostData(const uint32_t &token, std::vector<RsGxsChannelPost> &posts, std::vector<RsGxsComment> &cmts) = 0;
+
+	RS_DEPRECATED_FOR(getChannelsContent)
+	virtual bool getPostData(const uint32_t &token, std::vector<RsGxsChannelPost> &posts) = 0;
+
+	/**
+	 * @brief toggle message read status
+	 * @deprecated
+	 * @param[out] token GXS token queue token
+	 * @param[in] msgId
+	 * @param[in] read
+	 */
+	RS_DEPRECATED_FOR(markRead)
+	virtual void setMessageReadStatus(
+	        uint32_t& token, const RsGxsGrpMsgIdPair& msgId, bool read) = 0;
 
 	/**
 	 * @brief Share channel publishing key
 	 * This can be used to authorize other peers to post on the channel
-	 * @jsonapi{development}
+	 * @deprecated
 	 * @param[in] groupId Channel id
 	 * @param[in] peers peers to which share the key
 	 * @return false on error, true otherwise
 	 */
+	RS_DEPRECATED_FOR(shareChannelKeys)
 	virtual bool groupShareKeys(
 	        const RsGxsGroupId& groupId, const std::set<RsPeerId>& peers ) = 0;
 
 	/**
 	 * @brief Request subscription to a group.
 	 * The action is performed asyncronously, so it could fail in a subsequent
 	 * phase even after returning true.
-	 * @jsonapi{development}
+	 * @deprecated
 	 * @param[out] token Storage for RsTokenService token to track request
 	 * status.
 	 * @param[in] groupId Channel id
 	 * @param[in] subscribe
 	 * @return false on error, true otherwise
 	 */
+	RS_DEPRECATED_FOR(subscribeToChannel)
 	virtual bool subscribeToGroup( uint32_t& token, const RsGxsGroupId &groupId,
 	                               bool subscribe ) = 0;
 
 	/**
 	 * @brief Request channel creation.
 	 * The action is performed asyncronously, so it could fail in a subsequent
 	 * phase even after returning true.
+	 * @deprecated
 	 * @param[out] token Storage for RsTokenService token to track request
 	 * status.
 	 * @param[in] group Channel data (name, description...)
 	 * @return false on error, true otherwise
 	 */
+	RS_DEPRECATED_FOR(createChannel)
 	virtual bool createGroup(uint32_t& token, RsGxsChannelGroup& group) = 0;
 
 	/**
 	 * @brief Request post creation.
 	 * The action is performed asyncronously, so it could fail in a subsequent
 	 * phase even after returning true.
+	 * @deprecated
 	 * @param[out] token Storage for RsTokenService token to track request
 	 * status.
 	 * @param[in] post
 	 * @return false on error, true otherwise
 	 */
+	RS_DEPRECATED
```

Comment:
No _FOR here?

Reply:
because the right version to use has the same name

Reply:
But, is it possible to get class handler too?
I see that for RsGxsIfaceHelper::getTokenService() which is deprecated for RsGenExchange::getTokenService(), but not easy to know...
One more thing, in /libretroshare/src/util/rsdeprecate.h, you don't take care of clang. It's why I don't have message FOR. ;)

Reply:
What is inside the parentheses of `RS_DEPRECATED_FOR()` get directly passed trough per-processor stringizer operator `#` and then printed, so you probably can put almost anything within those parentheses (test it please), about taking care of clang I haven't a comfortable clang based tool-chain to test it so please make a PR to take care of it ;)

Reply:
Where did you found __GNUC__ value starting using __deprecated__(message)?
Because here, with Clang v7.0.0, __GNUC__ == 4 && __GNUC_MINOR__ == 2 and message works:
/libretroshare/src/services/p3gxscircles.cc:1636: warning: 'getTokenService' is deprecated: Use 'RsGenExchange::getTokenService()' instead [-Wdeprecated-declarations]
GCC can handle it since 4.9:
https://gcc.gnu.org/gcc-4.9/changes.html#cxx

Reply:
Don't remember at all, anyway take in account that `GNUC`, ` GNUC_MINOR` ans similar defined by clang are mostly meaningless, clang has it's own way to check for version, and probably define those macro too for some kind of retro-compatibility, but we must not rely on those when compiling with clang.

Reply:
also the page you linked talks about  C++1y deprecated attribute which most probably is not supported when compiling in C++11 mode (which is how we compile)